internal/gps/vcs_repo_test.go

```diff
@@ -46,6 +52,53 @@ func TestErrs(t *testing.T) {
 	}
 }
 
+func TestNewCtxRepo(t *testing.T) {
```

Comment:
Let's split this into two tests, one for the simple happy path, and the other for the failure + recovery path.